examples/nirfmxspecan/spectrum-basic.py

```diff
@@ -0,0 +1,151 @@
+# This program demonstrates the use of RFmx SpecAn to read a spectrum. The steps are:
+# 1. Open a new RFmx session
+# 2. Configure the basic signal properties  (Center Frequency, Reference Level and External Attenuation)
+# 3. Configure Spectrum Span
+# 4. Configure Spectrum RBW filter
+# 5. Configure Spectrum Averaging
+# 6. Read Spectrum Measurement Results
+# 7. Close the RFmx Session
+#
+# The gRPC API is built from the C API. RFmx SpecAn documentation is installed with the driver at:
+# C:\Program Files (x86)\National Instruments\RFmx\SpecAn\Documentation\rfmxspecancvi.chm
+#
+# Getting Started:
+#
+# To run this example, install "RFmx SpecAn" on the server machine.
+# Link: # https://www.ni.com/en-us/support/downloads/software-products/download.rfmx-specan.html
+#
+# For instructions on how to use protoc to generate gRPC client interfaces, see our "Creating a gRPC Client" wiki page.
+# Link: https://github.com/ni/grpc-device/wiki/Creating-a-gRPC-Client
+#
+# Refer to the NI-RFmxSpecAn gRPC Wiki for the latest C Function Reference:
+# Link: https://github.com/ni/grpc-device/wiki/NI-RFmxSpecAn-C-Function-Reference
+#
+# Running from command line:
+#
+# Server machine's IP address, port number, and physical channel name can be passed as separate command line arguments.
+#   > python spectrum-basic.py <server_address> <port_number> <physical_channel_name>
+# If they are not passed in as command line arguments, then by default the server address will be "localhost:31763", with "SimulatedRFSA" as the resource name
+
+from math import floor
+import grpc
+import sys
+
+import nirfmxspecan_pb2 as nirfmxspecan_types
+import nirfmxspecan_pb2_grpc as grpc_nirfmxspecan
+
+server_address = "localhost"
+server_port = "31763"
+session_name = "NI-RFSASession"
+
+# Resource name and options for a simulated 5663 client.
+resource = "SimulatedRFSA"
+options = "Simulate=1,DriverSetup=Model:5663"
+
+# Read in cmd args
+if len(sys.argv) >= 2:
+    server_address = sys.argv[1]
+if len(sys.argv) >= 3:
+    server_port = sys.argv[2]
+if len(sys.argv) >= 4:
+    resource = sys.argv[3]
+    options = ""
+
+# Create a gRPC channel + client.
+channel = grpc.insecure_channel(f"{server_address}:{server_port}")
+client = grpc_nirfmxspecan.NiRFmxSpecAnStub(channel)
+instr = None
+
+
+def format_frequency(f):
+    # adapted from https://stackoverflow.com/a/1094933
+    suffix = "Hz"
+    for unit in ["", "k", "M", "G"]:
+        if abs(f) < 1000.0:
+            return f"{f:3.3f} {unit}{suffix}"
+        f /= 1000.0
+    return f"{f:.3f} G{suffix}"
+
+
+# Raise an exception if an error was returned
+def raise_if_error(response):
+    if response.status != 0:
+        error_response = client.GetError(
+            nirfmxspecan_types.GetErrorRequest(
+                instrument_handle=instr,
```

Comment:
(Not necessarily for this PR): This `instrument_handle` name seems too long. The obvious thing is to drop handle and go with just `instrument`, which is what we did for DAQ `taskHandle`s. That seems pretty good. `instr` is appealing too, but it's better to err on the side of complete words these days. Any other ideas?

Reply:
Yeah, I like `instrument` - filed #437.